pkg/controller/infinispan/util/cluster.go

```diff
@@ -90,5 +92,96 @@ func (c Cluster) GetClusterMembers(secretName, podName, namespace, protocol stri
 
 		return health.ClusterHealth.Nodes, nil
 	}
+
 	return nil, fmt.Errorf("unexpected error getting cluster members, stderr: %v, err: %v", execErr, err)
 }
+
+func (c Cluster) ExistsCache(cacheName, secretName, podName, namespace, protocol string) bool {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return false
+	}
+
+	pass, err := c.GetPassword("operator", secretName, namespace)
+	if err != nil {
+		return false
+	}
+
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches/%s?action=config", protocol, podIP, cacheName)
+	commands := []string{"curl",
+		"-o", "/dev/null", "-w", "%{http_code}", // ignore output and get http status code
+		"-u", fmt.Sprintf("operator:%v", pass),
+		"--head",
+		httpURL,
+	}
+
+	execOptions := ExecOptions{Command: commands, PodName: podName, Namespace: namespace}
+	execOut, _, err := c.Kubernetes.ExecWithOptions(execOptions)
+	if err != nil {
+		return false
+	}
+
+	httpCode, err := strconv.ParseUint(execOut.String(), 10, 64)
+	if err != nil {
+		return false
+	}
+
+	switch httpCode {
+	case 200:
+		return true
+	default:
+		return false
+	}
+}
+
+func (c Cluster) CreateCache(cacheName, cacheXml, secretName, podName, namespace, protocol string) error {
+	podIP, err := c.Kubernetes.GetPodIP(podName, namespace)
+	if err != nil {
+		return err
+	}
+
+	pass, err := c.GetPassword("operator", secretName, namespace)
+	if err != nil {
+		return err
+	}
+
+	httpURL := fmt.Sprintf("%s://%s:11222/rest/v2/caches/%s", protocol, podIP, cacheName)
+	commands := []string{"curl",
+		"-w", "\n%{http_code}", // add http status at the end
+		"-d", fmt.Sprintf("%s", cacheXml),
+		"-H", "Content-Type: application/xml",
+		"-u", fmt.Sprintf("operator:%v", pass),
+		"-X", "POST",
+		httpURL,
+	}
+
+	execOptions := ExecOptions{Command: commands, PodName: podName, Namespace: namespace}
+	execOut, execErr, err := c.Kubernetes.ExecWithOptions(execOptions)
+	if err != nil {
+		return fmt.Errorf("unexpected error creating cache, stderr: %v, err: %v", execErr, err)
+	}
+
+	// Reverse sort the standard output so that HTTP status code is first
+	execOutLines := strings.Split(execOut.String(), "\n")
+	reverse(execOutLines)
+
+	httpCode, err := strconv.ParseUint(execOutLines[0], 10, 64)
```

Comment:
Not a bug but
seems you only use the last execOutLines string. Do you really need to reverse the array?

Reply:
Yeah, that's on purpouse. The curl as defined shows the output and then a line break and the HTTP status code. So, to get the status code I split the entire output by line and reverse it. Then, the first element would be the HTTP status code alone.

Reply:
The output of the command can itself have line breaks.